packages/block-library/src/embed/index.js

```diff
@@ -183,10 +191,32 @@ export function getEmbedEdit( title, icon ) {
 				}
 
 				if ( aspectRatioClassName ) {
-					const className = classnames( this.props.attributes.className, 'wp-has-aspect-ratio', aspectRatioClassName );
-					this.props.setAttributes( { className } );
+					return {
+						[ aspectRatioClassName ]: true,
+						'wp-has-aspect-ratio': true,
+					};
 				}
 			}
+
+			return {};
+		}
+
+		/**
+		 * Maybe sets the appropriate CSS class names to enforce an aspect ratio when the embed
+		 * is resized if the HTML has an iframe with width and height set.
+		 *
+		 * @param {string} html The preview HTML that possibly contains an iframe with width and height set.
+		 */
+		maybeSetAspectRatioClassName( html ) {
+			const { allowResponsive } = this.props.attributes;
```

Comment:
I think we should update the attribute name as well to reflect the text in the UI. This could be `allowVideoResizing` or something like that.

Reply:
Embeds can be video, audio, slideshows, etc. I think videos are likely to be the most common case where responsiveness is desired but they aren't the only case for sure.

Reply:
Perhaps we say "videos and other content" or something like that. Slideshows that embed in a fixed size iframe would be handled by this too.

Reply:
"Videos and other content" seems great; videos are probably the main use case so makes sense to highlight them.

Maybe the attribute could be `allowResizing` in that case.

Reply:
I'm not sure about `allowResizing` because that could suggest that we're allowing the user to resize the content, and we're actually enabling responsive content.

Reply:
Hmmm, fair enough. Okay, `allowResponsive` as an attribute it is!

My only issue now I think is around that `maybe` function, I've left comments there.